var/spack/repos/builtin/packages/nalu-wind/package.py

```diff
@@ -117,4 +117,12 @@ def cmake_args(self):
         if 'darwin' in spec.architecture:
             options.append('-DCMAKE_MACOSX_RPATH:BOOL=ON')
 
+        if '--run-tests' or '--test' in spec:
+            options.append('-DENABLE_TESTS:BOOL=ON')
+
         return options
+
+    @run_after('build')
+    @on_package_attributes(run_tests=True)
+    def test(self):
+        make('test')
```

Comment:
This should already be present in the `CMakePackage` base class, right?

Reply:
I'm really not sure.  I was just looking at what perl does. Will you provide an example/explain further?

Reply:
ah I see.  I was trying this before without the test function and it wasn't running the tests.  I will double check on another system though.  Looks like we don't call the CMakePackage::check() function though.  Does it need to be explicitly called in the app package.py?

Reply:
`perl` is a special case because they have their own custom build system.

Try removing this function and running `spack install -v --test=root nalu-wind`. It _should_ run the check function if it finds a test or check target in the Makefile.

Reply:
@adamjstewart I remember the issue for this.  We store extra data in a submodule for our testsuite that is normally not included.  Modifying the package by adding the `submodules` command is not a desired behavior for normal installs.  Correct @jrood-nrel? @adamjstewart is there a way we can conditionally turn on the submodules based on if testing is requested during the install?

Reply:
If you mean the `submodules=True` entry in `version(...)` then no. But I'm sure there is a way to run it during the test function. It shouldn't be too bad to always download submodules though.

Reply:
This submodule contains meshes, so it adds about 1GB.

Reply:
In that case, I would just run the appropriate git command inside of the `test` method.

Reply:
Thanks for the suggestion @adamjstewart.  I think I found a good solution.  May have to add a unit test still b/c I added a new function. Not really sure what the criteria is for that.  If so were should I put it?